build_tools/cmake/iree_lit_test.cmake

```diff
@@ -75,26 +76,30 @@ function(iree_lit_test)
   iree_package_ns(_PACKAGE_NS)
   string(REPLACE "::" "/" _PACKAGE_PATH ${_PACKAGE_NS})
   set(_NAME_PATH "${_PACKAGE_PATH}/${_RULE_NAME}")
-  add_test(
+  list(APPEND _RULE_LABELS "${_PACKAGE_PATH}")
+
+  iree_add_installed_test(
     NAME
-      ${_NAME_PATH}
+      ${_RULE_NAME}
     COMMAND
       # We run all our tests through a custom test runner to allow setup
       # and teardown.
       "${CMAKE_SOURCE_DIR}/build_tools/cmake/run_test.${IREE_HOST_SCRIPT_EXT}"
       "${CMAKE_SOURCE_DIR}/iree/tools/run_lit.${IREE_HOST_SCRIPT_EXT}"
       ${_TEST_FILE_PATH}
       ${_DATA_DEP_PATHS}
-    WORKING_DIRECTORY
-      "${CMAKE_BINARY_DIR}"
+    INSTALLED_COMMAND
+      # TODO: Make the lit runner be not a shell script and more cross-platform.
+      bin/run_lit.${IREE_HOST_SCRIPT_EXT}
+      ${_TEST_FILE_PATH}
+      ${_DATA_DEP_PATHS}
   )
-
-  list(APPEND _RULE_LABELS "${_PACKAGE_PATH}")
-  set_property(TEST ${_NAME_PATH} PROPERTY LABELS "${_RULE_LABELS}")
   set_property(TEST ${_NAME_PATH} PROPERTY REQUIRED_FILES "${_TEST_FILE_PATH}")
```

Comment:
An extra point for my comment about taking the test name instead. This is now separately computing the name of the test created above via the function.

Reply:
Ack.